src/components/AddComment.jsx

```diff
@@ -0,0 +1,57 @@
+import { useContext, useState } from "react";
+import { postComment } from "../../Utils/api";
+import { useParams, useNavigate } from "react-router-dom";
+import UserContext from "../Context/UserContext";
+
+const AddComment = ({ setComments }) => {
+  const navigate = useNavigate();
+  const { article_id } = useParams();
+  const [error, setError] = useState(null);
+  const [newComment, setNewComment] = useState("");
+  const { loggedInUser, isLoggedIn, handlePrevPath } = useContext(UserContext);
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    postComment(newComment, article_id, loggedInUser.username)
+      .then((commentFromApi) => {
+        setComments((currentComment) => [commentFromApi, ...currentComment]);
+        setNewComment("");
+        setError(null);
+      })
+      .catch((error) => {
+        setError("Something went wrong, please try again.");
+      });
+  };
+
+  return (
+    <div>
+      {isLoggedIn ? (
+        <form onSubmit={handleSubmit}>
+          <label>Add Comment</label>
+          <textarea
+            placeholder="Write your comment"
+            value={newComment}
+            onChange={(e) => setNewComment(e.target.value)}
+          ></textarea>
+          <br />
+          <br />
+          <br />
+          <button>Add Comment</button>
+        </form>
```

Comment:
The user can currently mash the post button and post loads of empty comments - can you add some form validation to this page please to prevent this